src/assets/scripts/client/aircraft/ModeControl/modeControlConstants.js

```diff
@@ -0,0 +1,49 @@
+export const MCP_ALTITUDE_MODES = {
+    HOLD: 'HOLD',
+    APPROACH: 'APPROACH',
+    LEVEL_CHANGE: 'LEVEL_CHANGE',
+    OFF: 'OFF',
+    VERTICAL_SPEED: 'VERTICAL_SPEED',
+    VNAV: 'VNAV'
+};
+
+export const MCP_SPEED_MODES = {
+    LEVEL_CHANGE: 'LEVEL_CHANGE',
+    N1: 'N1',
+    OFF: 'OFF',
+    HOLD: 'HOLD',
+    VNAV: 'VNAV'
+};
+
+export const MCP_HEADING_MODES = {
+    HOLD: 'HOLD',
+    LNAV: 'LNAV',
+    OFF: 'OFF',
+    VOR_LOC: 'VOR_LOC'
+};
+
+export const MCP_AUTOPILOT_MODES = {
+    ON: 'ON',
+    OFF: 'OFF'
+};
+
+export const MCP_MODES = {
+    ALTITUDE: MCP_ALTITUDE_MODES,
+    AUTOPILOT: MCP_AUTOPILOT_MODES,
+    HEADING: MCP_HEADING_MODES,
+    SPEED: MCP_SPEED_MODES
+};
+
+export const MCP_MODE_NAME = {
+    ALTITUDE: 'altitudeMode',
+    AUTOPILOT: 'autopilotMode',
+    HEADING: 'headingMode',
+    SPEED: 'speedMode'
+};
+
+export const MCP_PROPERTY_MAP = {
+    ALTITUDE: 'altitude',
+    AUTOPILOT: 'autopilot',
+    HEADING: 'heading',
```

Comment:
the term _fields_, in the front end space, is more often a reference to forms and inputs.  although it may be an acceptable aviation term, it is not a good name here because the values are not fields.  

The name actually leads you to its use.  `property` is a programming term.  This object is a map of mcp properties, where we map a, object key to a _property_ value.  that means we can do things like `this._modeController[MCP_PROPERTY_MAP.ALTITUDE]`.

The stuff in our object are actually _properties_ of the class.